client/src/main/java/io/split/client/SplitFactoryImpl.java

```diff
@@ -240,8 +240,10 @@ public SplitFactoryImpl(String apiToken, SplitClientConfig config) throws URISyn
                 _impressionsManager, _eventsTask, _telemetrySyncTask, _uniqueKeysTracker);
         SplitAPI splitAPI = SplitAPI.build(_httpclient, buildSSEdHttpClient(apiToken, config, _sdkMetadata));
 
+        SplitParser splitParser = new SplitParser();
```

Comment:
please don't create a new instance of `SplitParser`. Try to reuse the same one that you are using in the `SplitFetcher`.